sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/codegen/BufferHolder.java

```diff
@@ -61,6 +61,10 @@
    * Grows the buffer by at least neededSize and points the row to the buffer.
    */
   void grow(int neededSize) {
+    if (neededSize < 0) {
```

Comment:
I think we can use `assert` here. When `neededSize` is negative, there must be an overflow.